packages/fxa-content-server/app/scripts/lib/survey-filter.js

```diff
@@ -31,56 +35,189 @@ export const getConditionWithKey = (conditions, key) => {
   return conditions[key];
 };
 
-export const createConditionCheckFn = (valSourceCheck) => (comparator) => (
-  key
-) => (conds, valSource) => {
+/**
+ * Create a function that takes a survey's configured conditions and a function (f) that fetches
+ * the value to be compared to a condition specified by the 'key' param.
+ *
+ * @param {function} valSource a function that takes a function (a) and a comparator and apply the
+ *                             return value of (a) as the first argument to the comparator
+ * @param {function} comparator a function that compares two values and returns a boolean
+ * @param {string} key the configuration key where the condition value can be found
+ * @returns {function} a function that takes conditions and (f) (see description above) and
+ *                     returns a boolean
+ */
+export const createConditionCheckFn = (valSource) => (comparator) => (key) => (
+  conds,
+  fetchFn
+) => {
   // This turned out a little janky.  But we want to return true early if the
   // condition is not in the configs.
   const condVal = getConditionWithKey(conds, key);
   if (condVal === NONE) {
     return true;
   }
-  return valSourceCheck(valSource)(comparator)(condVal);
+  return valSource(fetchFn)(comparator)(condVal);
 };
 
-export const createAsyncConditionCheckFn = (valSourceCheck) => (comparator) => (
+// Like the function above but creates an async function because (f) can be async.
+export const createAsyncConditionCheckFn = (valSource) => (comparator) => (
   key
-) => async (conds, valSource) => {
+) => async (conds, fetchFn) => {
   const condVal = getConditionWithKey(conds, key);
   if (condVal === NONE) {
     return true;
   }
-  return (await valSourceCheck(valSource)(comparator))(condVal);
+  return (await valSource(fetchFn)(comparator))(condVal);
+};
+
+/**
+ * Creates a function that construct a UserAgent object when the given window
+ * object has a userAgent string and caches the UserAgent instance.
+ */
+export const createFetchUaFn = (window) => {
+  let ua;
+
+  return () => {
+    if (!window || !window.navigator || !window.navigator.userAgent) {
+      return undefined;
+    }
+
+    ua = ua || new UserAgent(window.navigator.userAgent);
```

Comment:
It would be defined after the first call (assuming happy path) to the function returned by `createFetchUaFn`, since it's in the closure of that function.